qiskit/transpiler/preset_passmanagers/level0.py

```diff
@@ -58,15 +64,24 @@ def level_0_pass_manager(pass_manager_config: PassManagerConfig) -> PassManager:
     basis_gates = pass_manager_config.basis_gates
     coupling_map = pass_manager_config.coupling_map
     initial_layout = pass_manager_config.initial_layout
+    layout_method = pass_manager_config.layout_method or 'dense'
+    routing_method = pass_manager_config.routing_method or 'stochastic'
     seed_transpiler = pass_manager_config.seed_transpiler
 
-    # 1. Use trivial layout if no layout given
+    # 1. Choose an initial layout if not set by user (default: trivial layout)
     _given_layout = SetLayout(initial_layout)
 
     def _choose_layout_condition(property_set):
         return not property_set['layout']
 
-    _choose_layout = TrivialLayout(coupling_map)
+    if layout_method == 'trivial':
```

Comment:
yeah that would be good to avoid repetition but right now the layout passes don't have the same signature, so I can't just pass the class here. 
```python
    if layout_method == 'trivial':
        _choose_layout = TrivialLayout(coupling_map)
    elif layout_method == 'dense':
        _choose_layout = DenseLayout(coupling_map, backend_properties)
    elif layout_method == 'noise_adaptive':
        _choose_layout = NoiseAdaptiveLayout(backend_properties)
```
It would be ideal to provide a unified interface though. I think in general there is too much repetition between the preset passmanagers, so a refactoring that makes them derive the same base and tweak would be great.